frontend/src/modules/Emailing/Components/Emailing.tsx

```diff
@@ -0,0 +1,207 @@
+    1.5 get other info the email needs to send:  
+       
+    */
+  const emailBody = "templates.get('template body')"
+  const emailSubject = "templates.get('template subject line')"
+  const emailRcpts = [
+    'zhanliam21@gmail.com',
+    'wz282@cornell.edu',
+    'willzhang21@icloud.com',
+  ]
+
+  const sendEmail = async () => {
+    // 2. obtaining auth token from local storage
+    const msAuthToken = localStorage.getItem('authToken') || ' '
+    console.log('access tok is ' + msAuthToken)
+
+    // 3. request to the backend to send mail
+    axios({
+      method: 'post',
+      url: `${API_ROOT}/email/send`,
+      data: {
+        authToken: msAuthToken,
+        emailAddress: email,
+        emailBody: emailBody,
+        emailSubject: emailSubject,
+        emailRcpts: emailRcpts,
+      },
+    }).then((res) => {
+      // 4. reading response for success or failure
+      console.log(res)
+      if (res.data === 'Email send success.') {
+        setEmailSent(true)
+        setSendError(false)
+      } else {
+        setEmailSent(false)
+        setSendError(true)
+        console.log('email send error')
+      }
+    })
+  }
+
+  const Succ = () => {
+    return (
+      <Box
+        display="flex"
+        flex-direction="row"
+        align-items=" center"
+        justifyContent="space-between"
+      >
+        <Typography>
+          {' '}
+          {emailSent ? 'Email is sent' : 'Email not yet sent'}!{' '}
+        </Typography>
+      </Box>
+    )
+  }
+
+  const EmailErr = () => {
+    return (
+      <Box
+        display="flex"
+        flex-direction="row"
+        align-items=" center"
+        justifyContent="space-between"
+      >
+        <Typography>
+          {' '}
+          {sendError
+            ? 'Email failed to send. Please log back in and reauthenticate to send email.'
+            : 'No email send error'}{' '}
+          !{' '}
+          {sendError && (
+            <Button
+              onClick={() => {
+                adminSignIn().then(() => sendEmail())
+              }}
+            >
+              Try Again{' '}
+            </Button>
+          )}
+        </Typography>
+      </Box>
+    )
+  }
+
+  return (
+    <StyledContainer>
+      <StyledLogoWrapper>
+        <StyledLogo />
+        <StyledText> Zing </StyledText>
+      </StyledLogoWrapper>
+      <Box
+        display="flex"
+        flex-direction="row"
+        align-items=" center"
+        justifyContent="space-between"
+      >
+        <Button onClick={() => sendEmail()} sx={{ height: '40px', mt: '10px' }}>
+          Send Emails
+        </Button>
+      </Box>
+      <Succ />
+      <EmailErr />
+    </StyledContainer>
+  )
+}
+
+/* Returns (number)
+    SUCC -> 201 
+    FAIL -> 401  */
+export const sendEmail = async (
+  emailSubject: string,
+  emailRcpts: string[],
+  emailBody: any,
+  emailSent: boolean,
+  setEmailSent: (arg: boolean) => void,
+  retried: boolean,
+  setRetried: (arg: boolean) => void
+) => {
```

Comment:
Yeah I think to be more generalizable this function could just focus on the part where it gets the token from local storage. Might want to leave the user email stuff to a function input so it can be obtained in the same way that it's obtained elsewhere in the app